### DISCUSSION TITLE
use class throw error

### DISCUSSION THREAD

**Discussion by ckken** [pmndrs/valtio#791] [Bug report]
## Summary
[store](https://github.com/ckken/emp-zustand/blob/bc9a19d401f8dc3489c3a8440e854582171dad88/projects/emp-valtio/src/store.ts#L31) 
[app](https://github.com/ckken/emp-zustand/blob/bc9a19d401f8dc3489c3a8440e854582171dad88/projects/emp-valtio/src/App.tsx#L26)
![image](https://github.com/pmndrs/valtio/assets/3890513/e00af1d0-553f-4db4-adff-8c34a67ecd96)
must add bind otherwise throw 
![image](https://github.com/pmndrs/valtio/assets/3890513/8b68514b-7f4d-4dac-a2f6-3f48ef903a13)

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/valtio/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [ ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by dai-shi** [comment]
https://github.com/pmndrs/valtio/blob/13b40936dda6116be1f7f04219b2a3c23b847007/docs/api/basic/proxy.mdx#L95

It might be an unsupported pattern. `bind` seems tricky.
